reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/tests/MockServerTest.kt

```diff
@@ -19,21 +20,28 @@ abstract class MockServerTest : BaseTest() {
 
     companion object {
         private val mockServerStarted = AtomicBoolean(false)
-        protected val mockServerWrapper: MockWebServerWrapper = MockWebServerWrapper()
+        private val MOCK_SERVER_LOCK = Any()
+        protected var mockServerWrapper: MockWebServerWrapper = MockWebServerWrapper()
 
         @BeforeClass
         @JvmStatic
         fun setupTestSuite() {
             if (mockServerStarted.compareAndSet(false, true)) {
-                mockServerWrapper.start()
+                synchronized(MOCK_SERVER_LOCK) {
```

Comment:
Isn't test runner single-threaded?

Reply:
I am not 100% sure, my memory tells me that it is but just wanted to make sure to avoid any possible flackyness here. Should we remove the lock ?

Reply:
On the other hand I think it is safe to keep the lock. The instrumented tests are being executed in a different thread and the mockWebServer can be interacted with from the SDK uploader threads. I think it is better to have this here, it will not harm us anyway.